### ISSUE TITLE
Editor outputs errors when attempting to access arrays and objects inside arrays

### ISSUE THREAD

**Issue by otonashixav** [godotengine/godot#59023]
### Godot version

v4.0.alpha4.official [f47097973]

### System information

Windows 10

### Issue description

Attempting to access arrays and objects inside arrays causes the following error to be printed:
```
core/variant/variant_op.cpp:967 - Condition "ret.type != BOOL" is true. Returning: false
```
The error is logged for every character while typing a method name, or on typing the index when accessing a nested array index.

The errors only appear in the editor; no runtime errors occur. Typing seems to have no effect on whether the error occurs; `=`, `:=`, `: Array[Type] =` and `: Array =` do not affect whether the errors occur or not. 

### Steps to reproduce

This code is in the minimal reproduction. 
```gdscript
extends Node

func _ready() -> void:
	# Attempting to access an index of or call a method of a nested array or object causes error
	# core/variant/variant_op.cpp:967 - Condition "ret.type != BOOL" is true. Returning: false
	# This is true regardless of whether `=`, `:=`, `: Array[Type] =` or `: Array =` is used
	var nestedArray := [[0]]
	var nestedDict := [{0: 0}]
	var nestedString := [""]
	var nestedObject := [RefCounted.new()]
	
	# typing these out will output errors
	nestedArray[0].append(1) # every character typed causes the error
	nestedArray[0][0] # [] is fine, 0 causes the error
	nestedObject[0].get_class() # every character typed causes the error
	
	# no errors
	nestedArray[0]
	nestedDict[0][0]
	nestedDict[0].clear()
	nestedString[0].begins_with("test")

```

### Minimal reproduction project

[repro.zip](https://github.com/godotengine/godot/files/8231464/repro.zip)

**Comment by Calinou** [comment]
Is this a recent bug, or can you reproduce this on previous 4.0alphas? You can download them [here](https://downloads.tuxfamily.org/godotengine/4.0/).

**Comment by otonashixav** [comment]
I can reproduce it on the earliest pre-alpha ([4.0-dev.20210727](https://downloads.tuxfamily.org/godotengine/4.0/pre-alpha/4.0-dev.20210727/)) and alpha3. I haven't tried the others.

**Comment by akien-mga** [comment]
Seems fixed as of 4.0 RC 5.
